src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,133 @@
+
+#include "rgw_tracer.h"
+
+using jConfig = jaegertracing::Config;
+using jLogger = jaegertracing::logging::Logger;
+
+namespace RGW_Tracing{
+
+    /*
+    //  Tracer
+    //
+    */
+    const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+    TracerConfiguration Tracer::tracer_conf;
+    
+    Tracer::Tracer(jConfig& conf):open_tracer(jaegertracing::Tracer::make(conf)){}
+    Tracer::Tracer(
+    std::string& service_name, 
+    const jConfig& conf, 
+    const std::shared_ptr<jLogger>& logger)
+    {
+        this->open_tracer = jaegertracing::Tracer::make(service_name, conf, logger);
+    }
+    Tracer::Tracer(std::string configFilePath){
+        auto configYAML = YAML::LoadFile(configFilePath);
+        auto config     = jaegertracing::Config::parse(configYAML);
+        this->open_tracer    = jaegertracing::Tracer::make(config);
+    }
+    Tracer::~Tracer(){
+        this->open_tracer->Close();
+    }
+    std::unique_ptr<Span> Tracer::start_trace(std::string trace_name){
+        if(is_enabled()) 
+            return std::unique_ptr<Span>(new Span(this->open_tracer->StartSpan(trace_name)));
+        else
+            return std::unique_ptr<Span>(new Span(this->noop_tracer->StartSpan(trace_name)));
+        
+    }
+    std::unique_ptr<Span> Tracer::start_span(std::string span_name, std::unique_ptr<Span>& parentSpan){
+        if(is_enabled()) 
+            return std::unique_ptr<Span>(new Span(this->open_tracer->StartSpan(span_name,{ parentSpan->get_reference() })));
+        else
+            return std::unique_ptr<Span>(new Span(this->noop_tracer->StartSpan(span_name)));
+    }
+    bool Tracer::is_enabled() const { return this->tracer_conf.is_enabled();}
+
+    /*
+    //  Span
+    //
+    */
+    Span::Span(std::unique_ptr<opentracing::Span> _span):span(_span.release()){};
+
+    Span::~Span(){ span->Finish();}
+
+    std::string Span::get_baggage(std::string key) const{
+        return this->span->BaggageItem(key);
+    }
+
+    opentracing::SpanReference Span::get_reference() const{
+        return opentracing::ChildOf(&this->span->context());
+    }
+
+    void Span::set_tag(std::string key, std::string value){
+        this->span->SetTag(key, value);
+        
+    }
+    void Span::set_log(std::string key, std::string value){
+        this->span->Log({{key, value}});
+    }
+    void Span::set_baggage(std::string key, std::string value){
+        this->span->SetBaggageItem(key, value);
+        
+
+    }
+
+    /*
+    //  Tracer Configuration
+    //
+    */
+    TracerConfiguration::TracerConfiguration(){
```

Comment:
no need to implement default ctor if its the only one. just remove from here and header